github/with_appengine.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This file provides glue for making github work on App Engine.
+
+// +build appengine
+
+package github
+
+import (
+	"context"
```

Comment:
If that's the case, that's going to be an issue. Just about every single file in package `github` imports `context`.

- https://gotools.org/github.com/google/go-github/github?rev=33bd2319b7816a4781538fb06f712d054d4c5271#activity.go-L8
- https://gotools.org/github.com/google/go-github/github?rev=33bd2319b7816a4781538fb06f712d054d4c5271#activity-events.go-L9
- https://gotools.org/github.com/google/go-github/github?rev=33bd2319b7816a4781538fb06f712d054d4c5271#activity-notifications.go-L9
- (60+ more files)

- https://github.com/google/go-github/blob/33bd2319b7816a4781538fb06f712d054d4c5271/github/activity.go#L8
- https://github.com/google/go-github/blob/33bd2319b7816a4781538fb06f712d054d4c5271/github/activity_events.go#L9
- https://github.com/google/go-github/blob/33bd2319b7816a4781538fb06f712d054d4c5271/github/activity_notifications.go#L9
- (60+ more files)

If we need appengine to import `golang.org/x/net/context` instead, we will have to make a copy of each and every .go file.

That would be a pretty significant disturbance, and it was one of the main reasons we decided to wait for Go 1.8 to come out, so that we could drop support for 1.6 and older before adding context support to this package.